cmd/notification-service/main.go

```diff
@@ -2,66 +2,55 @@ package main
 
 import (
 	"log"
-
-	"time"
 	"net/http"
 	"os"
-	"github.com/r3labs/sse"
 	"fmt"
-)
-
-func failOnError(err error, msg string) {
-	if err != nil {
-		log.Fatalf("%s: %s", msg, err)
-	}
-}
-
-func checkEurekaService(eurekaUp bool) bool {
-	duration := time.Duration(15) * time.Second
-	time.Sleep(duration)
-	url := "http://discoveryservice:8761/eureka/"
-	log.Println("Sending request to Eureka, waiting for response...")
-	request, _ := http.NewRequest("GET", url, nil)
-	request.Header.Set("Content-Type", "application/json")
+	"github.com/AITestingOrg/notification-service/internal/rabbitMQ"
+	"github.com/AITestingOrg/notification-service/internal/eureka"
+	"github.com/AITestingOrg/notification-service/internal/model"
 
-	client := &http.Client{}
-	response, responseErr := client.Do(request)
-	if responseErr != nil {
-		log.Printf("No response from Eureka, retrying...")
-		return false
-	}
-	if response.Status != "204 No Content" {
-		log.Printf("Success, Eureka was found!")
-		return true
-	}
-	return false
-}
+	"github.com/r3labs/sse"
+	"github.com/streadway/amqp"
+	"encoding/json"
+)
 
 func main() {
+	log.Println("Checking eureka")
 	localRun := false
 	if os.Getenv("EUREKA_SERVER") == "" {
 		localRun = true
 	}
 	if !localRun {
 		eurekaUp := false
-		log.Println("Waiting for Eureka...")
+		log.Println("Waiting for eureka...")
 		for eurekaUp != true {
-			eurekaUp = checkEurekaService(eurekaUp)
+			eurekaUp = eureka.CheckEurekaService()
 		}
 	}
 
-
-	server := sse.New()
-	server.CreateStream("messages")
+	log.Println("Creating server")
+	server := rabbitMQ.CreateServer("messages")
 
 	go func() {
-		server.Publish("messages", &sse.Event{
-			Data:  []byte("beat"),
-			Event: []byte("1"),
-		})
-	}()
+		rabbitMQ.InitializeConsumer(func(m amqp.Delivery){
+			log.Printf("Received message")
 
+			data, err := json.Marshal(model.Message{RoutingKey: m.RoutingKey, Body: m.Body})
 
+			if err != nil {
+				log.Printf("Parsing json failed")
+				return
+			}
+
+			server.CreateStream(m.UserId)
```

Comment:
The userId of the message is per Rabbit not per the system we are building. The correct user id should be in a field of the JSON message we should be receiving from RabbitMQ.